content/posts/eks-vs-ecs.md

```diff
@@ -0,0 +1,107 @@
+
+## What are ECS and EKS?
+
+These are both services from AWS which deal with container orchestration. Containers have become a crucial component in the modern tech stack, but as your applications scale, having an orchestrator is a must. 
+
+### ECS
+
+Amazon Elastic Container Service is a platform which will orchestrate your containers in a straightforward manner. You create an application definition and it will ensure that the container is running. 
+
+With ECS there are a couple of ways you can run your containers: on EC2 instances or with [Fargate](https://aws.amazon.com/fargate/), AWS's serverless compute-engine.
+
+With EC2 you operate like you normally would and configure auto-scaling groups, networking, etc. Fargate on the other hand abstracts away the individual server and has you only worry about cpu and memory needs. The cost will be higher per compute/memory resource but it handles a lot for you.
+
+
+### EKS
+
+Amazon Elastic Kubernetes Service is AWS's managed Kubernetes platform. Kubernetes has become the community-favored container orchestrator and for good reason. It's incredibly powerful, resilient, and now has a great ecosystem built around it.
+
+With all of the power of Kubernetes comes some complexity on the infrastructure side. Setting up and maintaining your own Kubernetes cluster can be challenging even for those with lots of experience. That's why whenever possible, we would recommend you use a managed Kubernetes service like EKS, GKE, or AKS since they solve those challenges for you.
+
+In December 2019, AWS started to [support using EKS with Fargate](https://aws.amazon.com/about-aws/whats-new/2019/12/run-serverless-kubernetes-pods-using-amazon-eks-and-aws-fargate/), so that specific differentiator between using ECS and EKS has vanished. Now you can deploy services in Kubernetes without worrying about the underlying infrastructure at all by using the Fargate serverless compute engine with EKS.
+
+Using EKS is in many ways similar to using ECS since whether you use Fargate or EC2, AWS will take care of provisioning the resources for you, letting you instead focus on the applications you will be running. The real differences come down to which container orchestrator you are using. ECS's solution or Kubernetes'.
+
+## Battle of the orchestrators
+
+If at the end of the day an application is going to be running in a container does it really matter which orchestrator you use as long as it works? Especially if one is simpler than the other? I would argue that it does matter. 
+
+Container orchestrators have become the new platform on which you build and deploy applications. While we often talk about vendor lock-in as a situation many people get into, it happens just as frequently with technologies. For instance, if you are developing for Linux machines your application isn't necessarily portable to a Windows machine. At least not without extra work. This is a form of lock-in.
```

Comment:
"about vendor lock-in as a situation many people get into" to "about avoiding vendor lock-in as a best practice"

"it happens just as frequently with technologies" - "with other technologies."?